src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -387,8 +387,7 @@ void DefaultOTARequestor::DisconnectFromProvider()
     }
 
     auto providerNodeId = GetProviderScopedId();
-    mCASESessionManager->FindExistingSession(providerNodeId)->Disconnect();
-    mCASESessionManager->ReleaseSession(providerNodeId);
+    mServer->GetSecureSessionManager().MarkSessionAsDefunct(providerNodeId, MakeOptional(Transport::SecureSession::Type::kCASE));
```

Comment:
This isn't super great, because the method as defined will only mark the most recently active session as defunct, which _may_ not be the session used by the OTA requestor before we got to this method.

To be safe, `MarkSessionAsDefunct` needs to be pivted to `MarkSessionsAsDefunct` and indiscriminately mark all sessions as defunct, but that's honestly too broad a cut and will affect potentially good and active sessions as well.

We should file a follow-up on how best to fix this, which may require the OTA requestor to actually track the session it's using by stashing away the `SessionHandle` in a `SessionHolder`.